pkg/kubelet/kubelet.go

```diff
@@ -2967,8 +3015,18 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			// If a resize is in progress, make sure the cache has the correct state in case the Kubelet restarted.
 			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
 		} else {
+			// podResizeConditionsOriginalLen is used to save original resize condition
+			podResizeConditionsOriginalLen := len(kl.statusManager.GetPodResizeConditions(pod.UID))
```

Comment:
adding this here just so it doesn't get lost but make sure you get an answer from @tallclair on https://github.com/kubernetes/kubernetes/pull/130387#discussion_r2074024203 on the desired behavior